### ISSUE TITLE
New iOS examples and fix dated iOS examples

### ISSUE THREAD

**Issue by ofTheo** [openframeworks/openFrameworks#7678]
Some of the old native UI examples didn't work anymore so they were removed for 0.12.0 
It would be great to do an audit of current examples and see what works, what doesn't and what is missing. 

Needed: 
- [ ] example showing OF running with native UI buttons 
- [ ] check video grabber example 
- [ ] check sound examples 
- [ ] check 3D / Assimp examples and make up to date with macOS ones 
- [ ] check for any examples that have non-ARC code ( we have ARC enabled by default now ).

**Comment by artificiel** [comment]
what about simulator support? it would facilitate a refactor of the examples. 

in https://github.com/openframeworks/openFrameworks/issues/6966#issuecomment-1324265560  @danoli3 says "Designed for iPad works"  but on M1 (13.4) I get this error (both with 0.12 and git-master):

<img width="810" alt="image" src="https://github.com/openframeworks/openFrameworks/assets/2172370/8561a4cb-7950-4e83-85ed-700b21e1eb9e">

**Comment by ofTheo** [comment]
Oh yes - @artificiel that's a separate known issue. I think will get fixed with @danoli3's changes. 
I think currently the only way to use the simulator is launching Xcode in Rosetta mode. 

This issue was just for updating the examples.

**Comment by artificiel** [comment]
ah yes sorry for not being clearer — I meant that having the simulator working would facilite this task, suggesting a form of priority, and reported that the work-around does not seem to be working.

NB I get the same error with a simulator build through rosetta on M1, launched with:
```
arch -x86_64 /Applications/Xcode.app/Contents/MacOS/Xcode
```

**Comment by danoli3** [comment]
Make sure on latest Xcode for Rosetta stuff: it should look like this
<img width="308" alt="Screenshot 2023-10-16 at 6 59 27 pm" src="https://github.com/openframeworks/openFrameworks/assets/830748/129a816f-7448-4b98-8ed3-33811d63e3b8">

**Comment by ofTheo** [comment]
@artificiel Rosetta would just mean x86_64 which you can see at the top in the menu bar, so I think that is what he meant.

**Comment by artificiel** [comment]
@ofTheo yes, I just wanted to state the Xcode GUI behaviour I have is not the same — in @danoli3 's screenshot the entries are marked Rosetta, which makes them clear to choose from. Here my entries are not marked, which makes it a trial-and-error thing to somehow "discover" there are rosetta ones.

so someone trying to use the simulator might not get @danoli3's described behaviour, and I wonder why the behaviour is different. but maybe it's moot as the proper simulator will soon be fixed?

**Comment by azuremous** [comment]
@artificiel 

> @ofTheo yes, I just wanted to state the Xcode GUI behaviour I have is not the same — in @danoli3 's screenshot the entries are marked Rosetta, which makes them clear to choose from. Here my entries are not marked, which makes it a trial-and-error thing to somehow "discover" there are rosetta ones.

You can find rosetta from Product > Destination > Destination Architectures > Show Rosetta Destinations

**Comment by azuremous** [comment]
@ofTheo 
I try to update iPhoneGuiExample using storyboard not nib. 

`
UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"MyGuiView" bundle:nil];
    myGuiViewController	=  [storyboard instantiateViewControllerWithIdentifier:@"MyGuiView"];`
    
<img width="1010" alt="Screenshot 2024-01-24 at 17 33 53" src="https://github.com/openframeworks/openFrameworks/assets/319589/a60e16a9-ba9e-428a-91df-685660d8f492">

**Comment by azuremous** [comment]
I'm currently trying to use iOS examples, but some of them are not working.
UIActionSheet does not work in iosOrientationExample.
 So, I modified the example and used UIAlertController instead of UIActionSheet. This means we no longer use the UIActionSheet detector. How about changing the file name? Naturally, we should update the interface name as well.

I will also make modifications to other iOS examples as much as I can.

**Comment by ofTheo** [comment]
Amazing @azuremous !!

I would say all the iOS examples are completely open to being changed / rewritten to be brought up to date. 
So definitely do what makes the most sense to you. 

Happy if you want to do per example Pull Requests or larger PRs for multiple examples, whatever works best for you. 

Thanks so much!
Great to get some energy on this!!

**Comment by artificiel** [comment]
status report as of 20250301: most examples works!

low hanging fruit:

- linking problems: #8316 , #8314 and #8315 

- src files missing: #8313 

- CoreLocation tweaks — PRs included: #8309 #8311 , but also #8310

- GLKit seems pointless? (kind of empty and no README) #8317

unsure:

 - iosNativeARCExample crashes in usage. it's a more complex app; getting it to run correctly is a bit of work. I don't know if it should stay in the broken condition. not sure if relevant as it seems to target the ARC which is now standard.

 - iosExternalDisplayExample: it mostly works (via USB-C HDMI adapter) but the UX is broken mostly due to the rotation ambiguity which is a known problem. I think it should stay because the 2nd output management does work, so depending on use case the demo problems might not be true problems (TLDR: the 2nd screen stuff works; the demo is buggy due to other issues)

also some observations, none dealbreaker but maybe while the project/template or build system is being tweaked:

 - #8305 
 - #8308 and #8312

**Comment by ofTheo** [comment]
@artificiel thanks for diving into these. 
I think we should either remove or fix the ARC example whichever is easier. 

Def would be good to fix the iosExternalDisplayExample that's a handy one! :) 

I added most of the issues you mentioned to #7588

**Comment by artificiel** [comment]
@oftheo PR #8324 for removal of ARC (any substantial native/modern iOS demo energy should go into a fresh one, ideally swiftUI).
